src/components/numeral-date/numeral-date.spec.tsx

```diff
@@ -276,47 +309,71 @@ describe("NumeralDate", () => {
         ["2201", false],
         ["2202", false],
       ])(
-        "renders internal validation when year field is blurred and has incorrect value",
+        "renders internal validation when year field is blurred and has incorrect value (%o)",
         (value, isValid) => {
           wrapper = renderWrapper({
-            defaultValue: null,
-            dateFormat: ["dd", "mm", "yyyy"],
+            defaultValue: undefined,
             [internalValidationProp]: true,
           });
           const input = wrapper.find("input").at(2);
           input.simulate("change", { target: { value } });
           input.simulate("blur");
           jest.runAllTimers();
           wrapper.update();
-          expect(wrapper.find(Textbox).at(2).props()[validationType]).toBe(
+
+          const dateTextboxes = wrapper.find(Textbox);
+          expect(
+            dateTextboxes.at(dateTextboxes.length - 1).props()[validationType]
+          ).toBe(
             isValid ? "" : "Year should be a number within a 1800-2200 range.\n"
           );
         }
       );
 
-      it("renders multiline internal validation when all fields are blurred with incorrect values", () => {
-        wrapper = renderWrapper({
-          defaultValue: null,
-          dateFormat: ["dd", "mm", "yyyy"],
-          [internalValidationProp]: true,
-        });
-        const dayInput = wrapper.find("input").at(0);
-        dayInput.simulate("change", { target: { value: "0" } });
-        dayInput.simulate("blur");
-        const monthInput = wrapper.find("input").at(1);
-        monthInput.simulate("change", { target: { value: "0" } });
-        monthInput.simulate("blur");
-        const yearInput = wrapper.find("input").at(2);
-        yearInput.simulate("change", { target: { value: "0" } });
-        yearInput.simulate("blur");
-        jest.runAllTimers();
-        wrapper.update();
-        expect(wrapper.find(Textbox).at(2).props()[validationType]).toBe(
-          "Day should be a number within a 1-31 range.\n" +
-            "Month should be a number within a 1-12 range.\n" +
-            "Year should be a number within a 1800-2200 range.\n"
-        );
-      });
+      describe.each<NumeralDateProps["dateFormat"][]>([
+        [["dd", "mm"]],
+        [["dd", "mm", "yyyy"]],
+      ])(
+        "when the format is set to %o and values are changed to be invalid",
+        (dateFormat) => {
+          it("multiline internal validation matching the format should be rendered", () => {
```

Comment:
What do you think about splitting this test in two - one for `["dd","mm"]` format and one for the `["dd", "mm", "yyyy"]` format? That way we won't need to add the conditional statements within the test.

Reply:
if I do that we'll end up with duplication of more than 30 lines of code instead of one conditional statement. 
Instead of that conditional I could move the expected message into a variable that is passed in the `.each` block, what do you think about that?

Reply:
Sounds good with me 👍🏼